dace/transformation/dataflow/redundant_array.py

```diff
@@ -97,4 +97,107 @@ def gnode(nname):
+            return False
+
+        # Make sure that the candidate is a transient variable
+        if not out_array.desc(sdfg).transient:
+            return False
+
+        # Make sure that both arrays are using the same storage location
+        if in_array.desc(sdfg).storage != out_array.desc(sdfg).storage:
+            return False
+
+        # Find occurrences in this and other states
+        occurrences = []
+        for state in sdfg.nodes():
+            occurrences.extend([
+                n for n in state.nodes() if isinstance(n, nodes.AccessNode)
+                and n.desc(sdfg) == out_array.desc(sdfg)
+            ])
+
+        if len(occurrences) > 1:
+            return False
+
+        # Only apply if arrays are of same shape (no need to modify memlet subset)
+        # if len(in_array.desc(sdfg).shape) != len(
+        #         out_array.desc(sdfg).shape) or any(i != o for i, o in zip(
+        #             in_array.desc(sdfg).shape,
+        #             out_array.desc(sdfg).shape)):
+        #     return False
+
+        return True
+
+    @staticmethod
+    def match_to_str(graph, candidate):
+        out_array = graph.nodes()[candidate[RedundantArray._out_array]]
+
+        return "Remove " + str(out_array)
+
+    def apply(self, sdfg):
+        def gnode(nname):
+            return graph.nodes()[self.subgraph[nname]]
+
+        graph = sdfg.nodes()[self.state_id]
+        in_array = gnode(RedundantArray._in_array)
+        out_array = gnode(RedundantArray._out_array)
+        memlet = graph.edges_between(in_array, out_array)[0].data
+        if memlet.data == in_array.data:
+            subset = memlet.subset
+        else:
+            subset = memlet.other_subset
+
+        for e in graph.out_edges(out_array):
+            # Modify all outgoing edges to point to in_array
+            path = graph.memlet_tree(e)
+            for pe in path:
+                if pe.data.data == out_array.data:
+                    pe.data.data = in_array.data
+                    if isinstance(subset, subsets.Indices):
+                        pe.data.subset.offset(subset, False)
+                    else:
+                        pe.data.subset = subset.compose(pe.data.subset)
+                elif pe.data.other_subset:
+                    if isinstance(subset, subsets.Indices):
+                        pe.data.other_subset.offset(subset, False)
+                    else:
+                        pe.data.other_subset = subset.compose(
+                            pe.data.other_subset)
+
+            # Redirect edge to out_array
+            graph.remove_edge(e)
+            graph.add_edge(in_array, e.src_conn, e.dst, e.dst_conn, e.data)
+
+        # Finally, remove in_array node
+        graph.remove_node(out_array)
+        # TODO: Should the array be removed from the SDFG?
+        # del sdfg.arrays[in_array]
+        if Config.get_bool("debugprint"):
+            RedundantArray._arrays_removed += 1
```

Comment:
This part can be removed. Also pointing to the wrong class.

Reply:
Fixed